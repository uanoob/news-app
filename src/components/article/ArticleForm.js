import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Send from '@material-ui/icons/Send';
import { checkValidityLength } from '../../utils';
import { createArticle, updateArticle } from '../../store/actions';

const styles = theme => ({
  container: {
    display: 'flex',
    flexDirection: 'column',
    flexWrap: 'wrap',
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
  },
  button: {
    margin: theme.spacing.unit,
  },
  rightIcon: {
    marginLeft: theme.spacing.unit,
  },
});

class ArticleForm extends Component {
  state = {
    title: '',
    titleInputTouched: false,
    titleInputValid: false,
    text: '',
    textInputTouched: false,
    textInputValid: false,
  };

  handleTitleInput = (e) => {
    this.setState({
      title: e.target.value,
      titleInputTouched: true,
      titleInputValid: checkValidityLength('title', e.target.value),
    });
  };

  handleTextInput = (e) => {
    this.setState({
      text: e.target.value,
      textInputTouched: true,
      textInputValid: checkValidityLength('text', e.target.value),
    });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const {
      articleId,
      onCreateArticle,
      onUpdateArticle,
      userId,
      userName,
      handleDialogClick,
    } = this.props;
    const { title, text } = this.state;
    if (!articleId) {
      onCreateArticle(title, text, userId, userName);
    } else {
      onUpdateArticle(articleId, title, text, articleId);
    }
    this.clearInputField();
    handleDialogClick();
  };

  handleCancelClick = () => {
    const { handleDialogClick } = this.props;
    this.clearInputField();
    handleDialogClick();
  };

  clearInputField = () => {
    this.setState({
      title: '',
      titleInputTouched: false,
      titleInputValid: false,
      text: '',
      textInputTouched: false,
      textInputValid: false,
    });
  };

  render() {
    const {
      classes,
      articleTitle,
      articleText,
      dialog,
      handleDialogClick,
      header,
      description,
    } = this.props;
    const {
      title,
      titleInputTouched,
      titleInputValid,
      text,
      textInputTouched,
      textInputValid,
    } = this.state;

    return (
      <Dialog
        open={dialog}
        onClose={handleDialogClick}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">{header}</DialogTitle>
        <DialogContent>
          <DialogContentText>{description}</DialogContentText>
          <TextField
            autoFocus
            id="article-title"
            label="Title article here"
            value={title || articleTitle}
            onChange={event => this.handleTitleInput(event)}
            className={classes.textField}
            margin="normal"
            variant="outlined"
            error={!titleInputValid && titleInputTouched}
            fullWidth
          />
          <TextField
            id="article-text"
            label="Text article here"
            multiline
            rows="10"
            value={text || articleText}
            onChange={event => this.handleTextInput(event)}
            className={classes.textField}
            margin="normal"
            variant="outlined"
            error={!textInputValid && textInputTouched}
            fullWidth
          />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.handleCancelClick} color="primary">
            Cancel
          </Button>
          <Button
            variant="contained"
            color="primary"
            className={classes.button}
            type="submit"
            onClick={this.handleSubmit}
            disabled={!titleInputValid}
          >
            Send
            <Send className={classes.rightIcon}>send</Send>
          </Button>
        </DialogActions>
      </Dialog>
    );
  }
}

ArticleForm.defaultProps = {
  articleId: '',
  articleTitle: '',
  articleText: '',
};

ArticleForm.propTypes = {
  classes: PropTypes.shape({
    container: PropTypes.string.isRequired,
    textField: PropTypes.string.isRequired,
  }).isRequired,
  articleId: PropTypes.string,
  onCreateArticle: PropTypes.func.isRequired,
  onUpdateArticle: PropTypes.func.isRequired,
  userId: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  dialog: PropTypes.bool.isRequired,
  handleDialogClick: PropTypes.func.isRequired,
  articleTitle: PropTypes.string,
  articleText: PropTypes.string,
  header: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  userId: state.user.user._id,
  userName: state.user.user.name,
});

const mapDispatchToProps = {
  onCreateArticle: createArticle,
  onUpdateArticle: updateArticle,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withStyles(styles)(ArticleForm));
