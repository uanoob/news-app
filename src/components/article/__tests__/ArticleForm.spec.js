import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import ArticleForm, { ArticleFormComponent } from '../ArticleForm';

describe('ArticleForm', () => {
  const mockCreateArticle = jest.fn();
  const mockUpdateArticle = jest.fn();
  const mockHandleDialogClick = jest.fn();
  const mockClearInputField = jest.fn();
  const mockHandleCancelClick = jest.fn();
  const initialState = {
    title: '',
    titleInputTouched: false,
    titleInputValid: false,
    text: '',
    textInputTouched: false,
    textInputValid: false,
  };
  const props = {
    classes: {
      container: '',
      textField: '',
      button: '',
      rightIcon: '',
    },
    articleId: '',
    articleTitle: '',
    articleText: '',
    onCreateArticle: mockCreateArticle,
    onUpdateArticle: mockUpdateArticle,
    handleDialogClick: mockHandleDialogClick,
    clearInputField: mockClearInputField,
    handleCancelClick: mockHandleCancelClick,
    userId: 'PropTypes.string.isRequired',
    userName: 'PropTypes.string.isRequired',
    dialog: false,
    header: 'PropTypes.string.isRequired',
    description: 'PropTypes.string.isRequired',
  };

  describe('ArticlesFormComponent initial', () => {
    const wrapper = shallow(<ArticleFormComponent {...props} />);

    it('render correctly ArticleFormComponent component', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('initialize ArticleFormComponent with initial state', () => {
      expect(wrapper.state()).toEqual(initialState);
    });
  });

  describe('ArticlesFormComponent checked input', () => {
    const wrapper = shallow(<ArticleFormComponent {...props} />);
    describe('valid title input', () => {
      const title = 'Enzyme';
      const event = {
        target: { name: 'title', value: title },
        preventDefault: () => {},
      };
      beforeEach(() => {
        wrapper.find('#article-title-input').simulate('change', event);
      });

      it('updates title field in state', () => {
        expect(wrapper.state('title')).toEqual(title);
      });
      it('updates titleInputTouched field in state', () => {
        expect(wrapper.state('titleInputTouched')).toEqual(true);
      });
      it('updates titleInputValid field in state', () => {
        expect(wrapper.state('titleInputValid')).toEqual(true);
      });
    });

    describe('invalid title input', () => {
      const title = 'd';
      const event = {
        target: { name: 'title', value: title },
        preventDefault: () => {},
      };
      beforeEach(() => {
        wrapper.find('#article-title-input').simulate('change', event);
      });

      it('updates title field in state', () => {
        expect(wrapper.state('title')).toEqual(title);
      });
      it('updates titleInputTouched field in state', () => {
        expect(wrapper.state('titleInputTouched')).toEqual(true);
      });
      it('updates titleInputValid field in state', () => {
        expect(wrapper.state('titleInputValid')).toEqual(false);
      });
    });

    describe('valid text input', () => {
      const text = 'Enzyme is a JavaScript Testing utility for React'
        + ' that makes it easier to assert, manipulate,'
        + 'and traverse your React Components output.';
      const event = {
        target: { name: 'text', value: text },
        preventDefault: () => {},
      };
      beforeEach(() => {
        wrapper.find('#article-text-input').simulate('change', event);
      });

      it('updates text field in state', () => {
        expect(wrapper.state('text')).toEqual(text);
      });
      it('updates textInputTouched field in state', () => {
        expect(wrapper.state('textInputTouched')).toEqual(true);
      });
      it('updates textInputValid field in state', () => {
        expect(wrapper.state('textInputValid')).toEqual(true);
      });
    });

    describe('invalid text input', () => {
      const text = 'Enzyme is a';
      const event = {
        target: { name: 'text', value: text },
        preventDefault: () => {},
      };
      beforeEach(() => {
        wrapper.find('#article-text-input').simulate('change', event);
      });

      it('updates text field in state', () => {
        expect(wrapper.state('text')).toEqual(text);
      });
      it('updates textInputTouched field in state', () => {
        expect(wrapper.state('textInputTouched')).toEqual(true);
      });
      it('updates textInputValid field in state', () => {
        expect(wrapper.state('textInputValid')).toEqual(false);
      });
    });
  });

  describe('ArticlesComponent submit', () => {
    const wrapper = shallow(<ArticleFormComponent {...props} />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'clearInputField');
    describe('when clicking the Send button and no articleId', () => {
      const event = { preventDefault: () => {} };
      wrapper.find('#article-submit-button').simulate('click', event);
      it('calls the onCreateArticle', () => {
        expect(mockCreateArticle).toBeCalledTimes(1);
      });
      it('calls clearInputField', () => {
        expect(instance.clearInputField).toBeCalledTimes(1);
      });
      it('set state to initialState', () => {
        expect(wrapper.state()).toEqual(initialState);
      });
    });
  });

  describe('ArticlesComponent submit + articleId', () => {
    const nextProps = {
      ...props,
      articleId: '123',
    };
    const wrapper = shallow(<ArticleFormComponent {...nextProps} />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'clearInputField');
    describe('when clicking the Send button and articleId', () => {
      const event = { preventDefault: () => {} };
      wrapper.find('#article-submit-button').simulate('click', event);
      it('calls the onUpdateArticle', () => {
        expect(mockUpdateArticle).toBeCalledTimes(1);
      });
      it('calls clearInputField', () => {
        expect(instance.clearInputField).toBeCalledTimes(1);
      });
      it('set state to initialState', () => {
        expect(wrapper.state()).toEqual(initialState);
      });
    });
  });
  describe('ArticleFormComponent cancel', () => {
    const wrapper = shallow(<ArticleFormComponent {...props} />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'clearInputField');
    describe('when clicking the Cancel button', () => {
      wrapper.find('#article-cancel-button').simulate('click');
      it('calls clearInputField', () => {
        expect(instance.clearInputField).toBeCalledTimes(1);
      });
    });
  });
  describe('ArticleFormComponent componentWillReceiveProps', () => {
    const nextProps = {
      ...props,
      articleTitle: 'Test title',
    };
    const wrapper = shallow(<ArticleFormComponent {...props} />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'componentWillReceiveProps');
    it('Component should call componentWillReceiveProps on update', () => {
      expect(instance.componentWillReceiveProps).toBeCalledTimes(0);
      wrapper.setProps(nextProps);
      expect(instance.componentWillReceiveProps).toBeCalledTimes(1);
    });
    it('update state title field', () => {
      expect(wrapper.state('title')).toEqual(nextProps.articleTitle);
    });
  });
  describe('ArticleForm mapStateToProps & mapDispatchToProps', () => {
    const mockStore = configureMockStore();
    const createArticle = () => ({
      type: 'CREATE_ARTICLE',
    });
    const updateArticle = () => ({
      type: 'UPDATE_ARTICLE',
    });
    const user = {
      user: {
        _id: 'testid',
        name: 'testname',
      },
    };
    const initialStateRedux = {
      user,
    };
    let store;
    let wrapper;
    beforeEach(() => {
      store = mockStore(initialStateRedux);
      wrapper = shallow(<ArticleForm store={store} />);
    });
    it('the state values were correctly passed as props', () => {
      expect(wrapper.props().store.getState()).toEqual(initialStateRedux);
    });
    it('the store dispatched the action CREATE_ARTICLE', () => {
      store.dispatch(createArticle());
      const actions = store.getActions();
      const expectedPayload = { type: 'CREATE_ARTICLE' };
      expect(actions).toEqual([expectedPayload]);
    });
    it('the store dispatched the action UPDATE_ARTICLE', () => {
      store.dispatch(updateArticle());
      const actions = store.getActions();
      const expectedPayload = { type: 'UPDATE_ARTICLE' };
      expect(actions).toEqual([expectedPayload]);
    });
  });
});
