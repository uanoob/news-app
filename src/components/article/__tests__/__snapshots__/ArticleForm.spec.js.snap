// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleForm ArticlesFormComponent initial render correctly ArticleFormComponent component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ArticleFormComponent
    articleId=""
    articleText=""
    articleTitle=""
    classes={
      Object {
        "button": "",
        "container": "",
        "rightIcon": "",
        "textField": "",
      }
    }
    clearInputField={[MockFunction]}
    description="PropTypes.string.isRequired"
    dialog={false}
    handleCancelClick={[MockFunction]}
    handleDialogClick={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    header="PropTypes.string.isRequired"
    onCreateArticle={
      [MockFunction] {
        "calls": Array [
          Array [
            "",
            "",
            "PropTypes.string.isRequired",
            "PropTypes.string.isRequired",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    onUpdateArticle={
      [MockFunction] {
        "calls": Array [
          Array [
            "123",
            "",
            "",
            "123",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    userId="PropTypes.string.isRequired"
    userName="PropTypes.string.isRequired"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-labelledby": "form-dialog-title",
      "children": Array [
        <WithStyles(DialogTitle)
          id="form-dialog-title"
        >
          PropTypes.string.isRequired
        </WithStyles(DialogTitle)>,
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)>
            PropTypes.string.isRequired
          </WithStyles(DialogContentText)>
          <TextField
            autoFocus={true}
            className=""
            error={false}
            fullWidth={true}
            id="article-title-input"
            label="Title article here"
            margin="normal"
            onChange={[Function]}
            required={false}
            select={false}
            value=""
            variant="outlined"
          />
          <TextField
            className=""
            error={false}
            fullWidth={true}
            id="article-text-input"
            label="Text article here"
            margin="normal"
            multiline={true}
            onChange={[Function]}
            required={false}
            rows="10"
            select={false}
            value=""
            variant="outlined"
          />
        </WithStyles(DialogContent)>,
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="primary"
            id="article-cancel-button"
            onClick={[Function]}
          >
            Cancel
          </WithStyles(Button)>
          <WithStyles(Button)
            className=""
            color="primary"
            disabled={true}
            id="article-submit-button"
            onClick={[Function]}
            type="submit"
            variant="contained"
          >
            Send
            <pure(SendIcon)
              className=""
            >
              send
            </pure(SendIcon)>
          </WithStyles(Button)>
        </WithStyles(DialogActions)>,
      ],
      "onClose": [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      },
      "open": false,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "PropTypes.string.isRequired",
          "id": "form-dialog-title",
        },
        "ref": null,
        "rendered": "PropTypes.string.isRequired",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(DialogContentText)>
              PropTypes.string.isRequired
            </WithStyles(DialogContentText)>,
            <TextField
              autoFocus={true}
              className=""
              error={false}
              fullWidth={true}
              id="article-title-input"
              label="Title article here"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              value=""
              variant="outlined"
            />,
            <TextField
              className=""
              error={false}
              fullWidth={true}
              id="article-text-input"
              label="Text article here"
              margin="normal"
              multiline={true}
              onChange={[Function]}
              required={false}
              rows="10"
              select={false}
              value=""
              variant="outlined"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "PropTypes.string.isRequired",
            },
            "ref": null,
            "rendered": "PropTypes.string.isRequired",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "className": "",
              "error": false,
              "fullWidth": true,
              "id": "article-title-input",
              "label": "Title article here",
              "margin": "normal",
              "onChange": [Function],
              "required": false,
              "select": false,
              "value": "",
              "variant": "outlined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "error": false,
              "fullWidth": true,
              "id": "article-text-input",
              "label": "Text article here",
              "margin": "normal",
              "multiline": true,
              "onChange": [Function],
              "required": false,
              "rows": "10",
              "select": false,
              "value": "",
              "variant": "outlined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(Button)
              color="primary"
              id="article-cancel-button"
              onClick={[Function]}
            >
              Cancel
            </WithStyles(Button)>,
            <WithStyles(Button)
              className=""
              color="primary"
              disabled={true}
              id="article-submit-button"
              onClick={[Function]}
              type="submit"
              variant="contained"
            >
              Send
              <pure(SendIcon)
                className=""
              >
                send
              </pure(SendIcon)>
            </WithStyles(Button)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancel",
              "color": "primary",
              "id": "article-cancel-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Send",
                <pure(SendIcon)
                  className=""
                >
                  send
                </pure(SendIcon)>,
              ],
              "className": "",
              "color": "primary",
              "disabled": true,
              "id": "article-submit-button",
              "onClick": [Function],
              "type": "submit",
              "variant": "contained",
            },
            "ref": null,
            "rendered": Array [
              "Send",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "send",
                  "className": "",
                },
                "ref": null,
                "rendered": "send",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-labelledby": "form-dialog-title",
        "children": Array [
          <WithStyles(DialogTitle)
            id="form-dialog-title"
          >
            PropTypes.string.isRequired
          </WithStyles(DialogTitle)>,
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)>
              PropTypes.string.isRequired
            </WithStyles(DialogContentText)>
            <TextField
              autoFocus={true}
              className=""
              error={false}
              fullWidth={true}
              id="article-title-input"
              label="Title article here"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              value=""
              variant="outlined"
            />
            <TextField
              className=""
              error={false}
              fullWidth={true}
              id="article-text-input"
              label="Text article here"
              margin="normal"
              multiline={true}
              onChange={[Function]}
              required={false}
              rows="10"
              select={false}
              value=""
              variant="outlined"
            />
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              id="article-cancel-button"
              onClick={[Function]}
            >
              Cancel
            </WithStyles(Button)>
            <WithStyles(Button)
              className=""
              color="primary"
              disabled={true}
              id="article-submit-button"
              onClick={[Function]}
              type="submit"
              variant="contained"
            >
              Send
              <pure(SendIcon)
                className=""
              >
                send
              </pure(SendIcon)>
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [MockFunction] {
          "calls": Array [
            Array [],
            Array [],
            Array [],
          ],
          "results": Array [
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
            Object {
              "isThrow": false,
              "value": undefined,
            },
          ],
        },
        "open": false,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "PropTypes.string.isRequired",
            "id": "form-dialog-title",
          },
          "ref": null,
          "rendered": "PropTypes.string.isRequired",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(DialogContentText)>
                PropTypes.string.isRequired
              </WithStyles(DialogContentText)>,
              <TextField
                autoFocus={true}
                className=""
                error={false}
                fullWidth={true}
                id="article-title-input"
                label="Title article here"
                margin="normal"
                onChange={[Function]}
                required={false}
                select={false}
                value=""
                variant="outlined"
              />,
              <TextField
                className=""
                error={false}
                fullWidth={true}
                id="article-text-input"
                label="Text article here"
                margin="normal"
                multiline={true}
                onChange={[Function]}
                required={false}
                rows="10"
                select={false}
                value=""
                variant="outlined"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "PropTypes.string.isRequired",
              },
              "ref": null,
              "rendered": "PropTypes.string.isRequired",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "className": "",
                "error": false,
                "fullWidth": true,
                "id": "article-title-input",
                "label": "Title article here",
                "margin": "normal",
                "onChange": [Function],
                "required": false,
                "select": false,
                "value": "",
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "error": false,
                "fullWidth": true,
                "id": "article-text-input",
                "label": "Text article here",
                "margin": "normal",
                "multiline": true,
                "onChange": [Function],
                "required": false,
                "rows": "10",
                "select": false,
                "value": "",
                "variant": "outlined",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                id="article-cancel-button"
                onClick={[Function]}
              >
                Cancel
              </WithStyles(Button)>,
              <WithStyles(Button)
                className=""
                color="primary"
                disabled={true}
                id="article-submit-button"
                onClick={[Function]}
                type="submit"
                variant="contained"
              >
                Send
                <pure(SendIcon)
                  className=""
                >
                  send
                </pure(SendIcon)>
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancel",
                "color": "primary",
                "id": "article-cancel-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  "Send",
                  <pure(SendIcon)
                    className=""
                  >
                    send
                  </pure(SendIcon)>,
                ],
                "className": "",
                "color": "primary",
                "disabled": true,
                "id": "article-submit-button",
                "onClick": [Function],
                "type": "submit",
                "variant": "contained",
              },
              "ref": null,
              "rendered": Array [
                "Send",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "send",
                    "className": "",
                  },
                  "ref": null,
                  "rendered": "send",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
